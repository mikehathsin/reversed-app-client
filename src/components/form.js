import { useState } from "react";
import Button from "react-bootstrap/Button";
import RBForm from "react-bootstrap/Form";

export const Form = ({ onSubmit, isLoading }) => {
  const [text, setText] = useState("");

  const handleChange = (event) => {
    setText(event.target.value);
  };

  return (
    <RBForm className="d-flex" onSubmit={onSubmit(text, setText)}>
      <RBForm.Control
        placeholder="Insert text"
        value={text}
        onChange={handleChange}
        disabled={isLoading}
      />
      <Button className="ms-2" type="submit" disabled={isLoading}>
        Send
      </Button>
    </RBForm>
  );
};
