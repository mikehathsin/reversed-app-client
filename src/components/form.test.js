import { render } from "@testing-library/react";

import { Form } from "./form";

describe("Form tests", () => {
  const handleSubmit = jest.fn();

  it("Should render correctly", () => {
    const component = render(<Form onSubmit={handleSubmit} />);
    component.getByPlaceholderText("Insert text");
    component.getByText("Send");
    expect(component.container).toMatchSnapshot();
  });
});
