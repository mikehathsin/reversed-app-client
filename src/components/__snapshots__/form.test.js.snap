// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form tests Should render correctly 1`] = `
<div>
  <form
    class="d-flex"
  >
    <input
      class="form-control"
      placeholder="Insert text"
      value=""
    />
    <button
      class="ms-2 btn btn-primary"
      type="submit"
    >
      Send
    </button>
  </form>
</div>
`;
